### PR TITLE
Enable cpp20

### PR DESCRIPTION
**Pull request by JasonMarechal25**

### CODE REVIEW

**Comment by flomnes** [comment]
Change the base branch ?

**Comment by JasonMarechal25** [comment]
> Change the base branch ?

Superfluous commit will be removed, as of now, it's more a test to see what happens in CI

**Comment by a-zakir** [comment]
Maybe an antares-deps compiled with std20

**Comment by a-zakir** [comment]
> Maybe an antares-deps compiled with std20
It may impact Xpansion…

**Comment by JasonMarechal25** [comment]
> > Maybe an antares-deps compiled with std20
> > It may impact Xpansion…

Depends if ABI is stable. Should be the case between 11 and 20

File: .github/workflows/centos7.yml [hunk 1]
```diff
@@ -4,8 +4,6 @@ on:
   release:
     types: [created]
   push:
-    branches:
```
Comment (sylvlecl) [inline .github/workflows/centos7.yml:7]: To be restored

**Review by sylvlecl — COMMENTED** [review]
Nice :) just a small fix in the centos CI is needed

File: src/ui/simulator/windows/hydro/management.cpp [hunk 2]
```diff
@@ -69,11 +68,8 @@ void Management::createComponents()
 
     const wxSize ourDefaultSize(55, wxDefaultSize.GetHeight());
 
-    enum
-    {
-        right = wxRIGHT | wxALIGN_RIGHT | wxALIGN_CENTER_VERTICAL,
-        left = wxLEFT | wxALIGN_LEFT | wxALIGN_CENTER_VERTICAL,
-    };
+    auto right = (wxRIGHT | wxALIGN_RIGHT | wxALIGN_CENTER_VERTICAL);
+    auto left = (wxLEFT | wxALIGN_LEFT | wxALIGN_CENTER_VERTICAL);
```
Comment (flomnes) [inline src/ui/simulator/windows/hydro/management.cpp]:
```suggestion
    const auto right = (wxRIGHT | wxALIGN_RIGHT | wxALIGN_CENTER_VERTICAL);
    const auto left = (wxLEFT | wxALIGN_LEFT | wxALIGN_CENTER_VERTICAL);
```

**Review by flomnes — APPROVED** [review]
LGTM. What about Windows ? No need to install a newer VS compiler ?

**Comment by flomnes** [comment]
Also make sure to document the fact that GCC 10+ is now required to build (C++17)

**Review by flomnes — APPROVED** [review]

**Pull request merged**